app/earn/[...vault]/info-blocks/asset-safety.tsx

```diff
@@ -1,36 +1,84 @@
 import CompactNumber from '@/components/compact-number/compact-number'
 import IconArrowRightShort from '@/components/icons/icon-arrow-right-short'
-import { getAssetSymbol } from '@/api/vaults/get-asset-symbol'
-import type { TokenSymbol } from '@/types'
+import { getAssetSymbol } from '@/api/protocol/vaults/get-asset-symbol'
+import { FormAction } from '@/store/slices/types'
 
 export function AssetSafety() {
-  const [amount] = useVaultDeposit()
-  const { vault } = useVaultContext()
+  const { inputValue = 0n, showPlaceholder, placeholderValue, vault } = useVaultContext()
+  const value = showPlaceholder ? placeholderValue : inputValue
+  const proportion = useVaultAssetProportion()
+
   return (
-    <CommonInfoBlock
-      title="Asset Safety"
-      items={[{
-        title: 'Portfolio',
-        value: 'Fully Protected',
-        icon: <IconArrowRightShort />,
-      }]}
-      description={<Amount amount={amount} decimals={vault.asset.decimals} symbol={getAssetSymbol(vault)} />}
-    />
+    <CommonInfoBlock>
+      <InfoBlockTitle>Asset Safety</InfoBlockTitle>
+
+      <InfoBlockList>
+        <InfoItem>
+          <InfoItemIcon><IconArrowRightShort /></InfoItemIcon>
+          <InfoItemTitle>Portfolio Rating</InfoItemTitle>
+          <InfoItemValue>
+            <PortfolioRating>{proportion}</PortfolioRating>
+          </InfoItemValue>
+        </InfoItem>
+      </InfoBlockList>
+
+      <SafetyDescription proportion={proportion} amount={value}>
+        <CompactNumber value={value} decimals={vault.asset.decimals} fractionDigits={2} hideTooltip>
+          &nbsp;{getAssetSymbol(vault)}
+        </CompactNumber>
+      </SafetyDescription>
+
+    </CommonInfoBlock>
   )
 }
 
-interface AmountProps {
-  amount: bigint
-  decimals: number
-  symbol: TokenSymbol
+function PortfolioRating({ children: proportion }: { children: number }) {
+  if (proportion === 1)
+    return 'Fully Protected'
+
+  if (proportion > 0)
+    return 'Partially Protected'
+
+  return 'Not Protected'
 }
 
-function Amount({ amount, decimals, symbol }: AmountProps) {
-  return (<>After deposit of <CompactNumber value={amount} decimals={decimals} fractionDigits={2} hideTooltip>
-    &nbsp;{symbol}
-  </CompactNumber>, the whole portfolio will be covered by insurance.
-  </>)
+function SafetyDescription({ children: displayAmount, proportion, amount }: PropsWithChildren<{ proportion: number; amount: bigint }>) {
+  const { formAction } = useVaultContext()
+
+  if (amount === 0n) {
+    return (
+      <InfoBlockDescription>
+        {proportion === 0
+          ? 'Whole your portfolio on the wallet, that not covered by insurance, but only assets in savings account can be recovered.'
+          : proportion === 1
+            ? 'Whole your portfolio in savings account is covered by insurance.'
+            : 'Only part of portfolio in savings account is covered by insurance.'
+        }
+      </InfoBlockDescription>
```

Comment:
Me and deepl cannot translate it, i assume it's grammatically incorrect:
> Whole your portfolio on the wallet, that not covered by insurance, but only assets in savings account can be recovered.

А тут вообще кажется, что мы частично страхуем то, что находится в ваулте:
> Only part of portfolio in savings account is covered by insurance.

```suggestion
      <InfoBlockDescription>
        {proportion === 0
          ? 'Your portfolio is not covered by insurance, only assets in a savings account can be recovered.'
          : proportion === 1
            ? 'Your entire portfolio is in a savings account and fully insured.'
            : 'Only the part of the portfolio that is in the savings account is covered by insurance.'
        }
      </InfoBlockDescription>
```

Reply:
You right, thnx, updated text